src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java

```diff
@@ -0,0 +1,67 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorStrategy;
+import pl.byczazagroda.trackexpensesappbackend.mapper.FinancialTransactionCategoryModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.service.FinancialTransactionCategoryService;
+import pl.byczazagroda.trackexpensesappbackend.service.FinancialTransactionCategoryServiceImpl;
+
+import java.util.List;
+
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = FinancialTransactionCategoryController.class,
+        excludeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE,
+                classes = FinancialTransactionCategoryServiceImpl.class),
+        includeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = {FinancialTransactionCategoryModelMapper.class, ErrorStrategy.class}))
+@ActiveProfiles("test")
+class FinancialTransactionCategoryControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private FinancialTransactionCategoryService financialTransactionCategoryService;
+
+    @Test
+    @DisplayName("when finding all financial transaction categories " +
+            "should return financial transaction category DTOs list and response status OK")
+    void shouldResponseStatusOKAndFinancialTransactionCategoryDTOsList() throws Exception {
+
+        // given
+        List<FinancialTransactionCategoryDTO> listDTOs = createListOfFinancialTransactionCategoryDTOs();
+        given(financialTransactionCategoryService.getFinancialTransactionCategories())
+                .willReturn(listDTOs);
+
+        // when
+
+        // then
+        mockMvc.perform(get("/api/categories")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.size()").value(3));
+
+    }
+
+    private List<FinancialTransactionCategoryDTO> createListOfFinancialTransactionCategoryDTOs() {
```

Comment:
Maybe this method can be private and static.

Reply:
done